### PR TITLE
Template editor JSON exporting and importing

### PR DESCRIPTION
**Pull request by zhang-yuanrui**
This PR fulfill 2 parts of functionalities:
1. Exporting templates with hierarchy into a JSON file by the interface `get_templates_as_json`
2. Importing a JSON file and build the templates from the JSON by the interface `load_templates`

Things not done yet in this PR:
1. Checking comprehensive user errors when loading a JSON file. I did some basic JSON format error checking but not exhaustive.
2. Prompting users if they want to rename when loading a root template with the same name as the existing one.

Please note that: in this PR, `load_templates` is supposed to be working when given a correct JSON input, ideally, the original / unchanged JSON file from `get_templates_as_json`

### CODE REVIEW

**Review by margalva — APPROVED** [review]

File: src/ansys/dynamicreporting/core/utils/report_remote_server.py [hunk 1]
```diff
@@ -972,6 +972,112 @@ def get_pptx_from_report(self, report_guid, directory_name=None, query=None):
...
+def _build_template_data(guid, templates_data, templates, template_guid_id_map):
+    curr_template = None
+    for template in templates:
+        if template.guid == guid:
+            curr_template = template
+    if curr_template is None:
+        return
+
+    curr_template_key = f"Template_{template_guid_id_map[curr_template.guid]}"
+    templates_data[curr_template_key] = {
```
Comment (viseshrp) [inline src/ansys/dynamicreporting/core/utils/report_remote_server.py:1053]: I would suggest you get the attributes from the object's class instead

File: src/ansys/dynamicreporting/core/utils/report_remote_server.py [hunk 2]
```diff
@@ -972,6 +972,112 @@ def get_pptx_from_report(self, report_guid, directory_name=None, query=None):
...
+        "name": curr_template.name,
+        "report_type": curr_template.report_type,
+        "date": curr_template.date,
```
Comment (viseshrp) [inline src/ansys/dynamicreporting/core/utils/report_remote_server.py:1056]: If you hardcode it this way, any new attribute will probably be missed out from this.

Comment (viseshrp) [reply src/ansys/dynamicreporting/core/utils/report_remote_server.py:1056]: Check the method from_db in `serverless/base.py` on how to get the fields.

**Review by viseshrp — CHANGES_REQUESTED** [review]

File: src/ansys/dynamicreporting/core/utils/report_remote_server.py (hunk 2 above)
Comment (viseshrp) [reply src/ansys/dynamicreporting/core/utils/report_remote_server.py:1056]: Here's another example https://github.com/ansys/pydynamicreporting/blob/82b592f1f49eea691c1b67cf3c1532ead1d67be0/src/ansys/dynamicreporting/core/serverless/base.py#L257

File: tests/conftest.py [hunk 3]
```diff
@@ -93,3 +93,29 @@ def adr_service_query(request, pytestconfig: pytest.Config) -> Service:
         tmp_service._container.save_config()
     tmp_service.start(create_db=False, exit_on_close=True, delete_db=False)
     return tmp_service
+
+
+@pytest.fixture
+def adr_template_json(request, pytestconfig: pytest.Config) -> Service:
+    use_local = pytestconfig.getoption("use_local_launcher")
```
Comment (viseshrp) [inline tests/conftest.py:100]: I would avoid creating new fixtures just to add a few templates. If you are only testing new template examples, please add them to the existing database and use the existing fixtures instead.

File: tests/conftest.py [hunk 4]
```diff
@@ -93,3 +93,29 @@ def adr_service_query(request, pytestconfig: pytest.Config) -> Service:
...
+    local_db = os.path.join("test_data", "template_json")
+    db_dir = os.path.join(request.fspath.dirname, local_db)
+    tmp_docker_dir = os.path.join(
+        os.path.join(request.fspath.dirname, "test_data"), "tmp_docker_json"
+    )
+    if use_local:
+        ansys_installation = pytestconfig.getoption("install_path")
+    else:
+        cleanup_docker(request)
```
Comment (viseshrp) [inline tests/conftest.py:109]: I merged an improvement to testing, so you may have to pull those and fix the merge conflicts. Also, fixtures are created slightly differently now. Check the new conftest.py.

**Review by viseshrp — CHANGES_REQUESTED** [review]

File: src/ansys/dynamicreporting/core/utils/report_remote_server.py [hunk 5]
```diff
@@ -1046,19 +1046,17 @@ def _build_template_data(guid, templates_data, templates, template_guid_id_map):
     for template in templates:
         if template.guid == guid:
             curr_template = template
-    if curr_template is None:
-        return
 
+    fields = ["name", "report_type", "date", "tags", "item_filter"]
```
Comment (zhang-yuanrui) [inline src/ansys/dynamicreporting/core/utils/report_remote_server.py:1050]: @viseshrp As I only need a subset of the attributes, I still hardcode over here but a bit better than previous way.

File: tests/test_report_remote_server.py [hunk 6]
```diff
@@ -348,8 +348,26 @@ def test_acls_start(request, get_exec) -> bool:
 
 
 @pytest.mark.ado_test
-def test_get_templates_as_json(adr_template_json) -> bool:
-    server = adr_template_json.serverobj
+def test_get_templates_as_json(adr_service_create) -> bool:
```
Comment (zhang-yuanrui) [inline tests/test_report_remote_server.py:351]: Use an existing fixture and populate templates manually.

**Pull request merged**